### PR TITLE
feat: #6662 dev portal fixes

### PR DESCRIPTION
**Pull request by willmcvay**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #6662
fixes: #6667 
fixes: #6518 
fixes: #6611

### CODE REVIEW

File: packages/developer-portal/src/components/analytics/costs/index.tsx [hunk 1]
```diff
@@ -78,7 +91,30 @@ export const AnalyticsCosts: FC = () => {
...
+          <ButtonGroup alignment="center">
+            <Button fixedWidth intent="secondary" onClick={closeModal}>
+              Close
+            </Button>
+          </ButtonGroup>
+        </Modal>
+      )}
+      <Subtitle>
+        Total for period: £
+        {aggregatedBilling.totalCost ? aggregatedBilling.totalCost.toFixed(2).padStart(2, '0') : '£0'}
```
Comment (bashleigh) [inline packages/developer-portal/src/components/analytics/costs/index.tsx:116]: pound sign already exists? So result for no totalCost would be `£ £0`

File: packages/developer-portal/src/components/analytics/costs/services-table.tsx [hunk 2]
```diff
@@ -17,15 +17,14 @@ export interface ServicesTableProps {
...
-  const services = billing?.services ?? []
+  const services = billing?.services?.filter((service) => service.name !== 'API Requests') ?? []
 
   return services.length ? (
     <Table
       className={elMb7}
       rows={
         services
           .map(({ name, amount, cost, itemCount, items }) => {
-            if (name === 'API Requests') return
```
Comment (bashleigh) [inline packages/developer-portal/src/components/analytics/costs/services-table.tsx:28]: this adds an additional loop. Backend I would say we could mitigate some milliseconds by preventing it but your call :) not a big issue.

File: packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx [hunk 3]
```diff
@@ -124,8 +124,8 @@ export const PipelineDeploymentTable: FC = () => {
               ({ type, created, buildStatus, buildVersion, currentlyDeployed, tasks }) => ({
                 cells: [
                   {
-                    label: 'Type',
-                    value: type ?? '',
+                    label: 'Type Release',
```
Comment (bashleigh) [inline packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx:127]: Release Type? Maybe? Release Method? Trigger Method?

File: packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx [hunk 4]
```diff
@@ -141,7 +141,13 @@ export const PipelineDeploymentTable: FC = () => {
                   },
                   {
                     label: 'Status',
-                    value: buildStatusToReadable(buildStatus as string),
+                    value: '',
+                    children: (
+                      <>
+                        <StatusIndicator intent={buildStatusToIntent(buildStatus as string)} />
+                        {buildStatusToReadable(buildStatus as string)}
+                      </>
```
Comment (bashleigh) [inline packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx:149]: Perhaps we should make this a reusable component as we've used it in many places

File: packages/developer-portal/src/components/apps/pipeline/pipeline-tasks-list.tsx [hunk 5]
```diff
@@ -27,8 +35,11 @@ export const TaskList: FC<TaskListProps> = ({ tasks }) => {
       </TableHeadersRow>
       {tasks.map((task) => (
         <TableRow key={task.id} className={fourColTable}>
-          <TableCell>{task.functionName}</TableCell>
-          <TableCell>{buildStatusToReadable(task.buildStatus as string)}</TableCell>
+          <TableCell>{buildStatusToReadable(task.functionName as string)}</TableCell>
```
Comment (bashleigh) [inline packages/developer-portal/src/components/apps/pipeline/pipeline-tasks-list.tsx:38]: function name is not a build status

**Review by bashleigh — CHANGES_REQUESTED** [review]
Potential issue with functionName values on tasks and duplicate pound sign

File: packages/developer-portal/src/components/apps/pipeline/pipeline-tasks-list.tsx (hunk 5 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/apps/pipeline/pipeline-tasks-list.tsx:38]: As discussed, just downcasing

File: packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx (hunk 4 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx:149]: Maybe? Not sure, only 2 places I think

File: packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx (hunk 3 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx:127]: Yep, will change

File: packages/developer-portal/src/components/analytics/costs/services-table.tsx (hunk 2 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/analytics/costs/services-table.tsx:28]: As discussed, only like 5 items absolute max

File: packages/developer-portal/src/components/analytics/costs/index.tsx (hunk 1 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/analytics/costs/index.tsx:116]: Yep, good spot

Comment (willmcvay) [reply packages/developer-portal/src/components/analytics/costs/index.tsx:116]: Yep, good spot

File: packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx (hunk 4 above)
Comment (bashleigh) [reply packages/developer-portal/src/components/apps/pipeline/pipeline-deployments-table.tsx:149]: quiet a few I've seen, doesn't matter we can do it another time :)

**Review by bashleigh — APPROVED** [review]

**Pull request merged**